pkg/storage/split/decider.go

```diff
@@ -165,10 +165,16 @@ func (d *Decider) MaybeSplitKey(now time.Time) roachpb.Key {
 		//
 		// We take the risk that the result may sometimes not be a good split
 		// point (or even in this range).
-		var err error
-		key, err = keys.EnsureSafeSplitKey(d.mu.splitFinder.Key())
-		if err != nil {
-			key = nil
+		//
+		// Note that we ignore EnsureSafeSplitKey when it returns an error since
```

Comment:
I don't get it - the comment above says that we don't want to allow mid-row splits and that SQL even crashes.. If this changed we should update that comment

Reply:
Oh, or we know for sure that in the error case the key is "in-between" rows? Seems like we should fix `EnsureSafeSplitKey` to not return an error at all, or at least explain this in its comment

Reply:
SQL sort of forces us to avoid mid-column family splits and we can't do so because you can't tell the column family suffix if you don't know the SQL schema. Also SQL issues plenty of scans that don't have the column family in the start key, so that's what the split finder sees. So when it sees a key that doesn't look like a SQL key, it has to assume that it doesn't cut columns in half, or it couldn't split most of the time (as we saw in the test this commit fixes). This probably holds but if you issued unfortunate batch requests you can definitely trick it into still splitting between col families.

I'm pretty unhappy about the situation in here and it seems reasonable to make SQL resilient to col families being split between ranges, not because I love that, but because I don't see any other way 😞

Reply:
Bleah. I would put these details in an issue and mention the issue in the comment.